src/CakeManager.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Migrations;
+
+use Phinx\Migration\Manager;
+
+/**
+ * Overrides Phinx Manager class in order to provide an interface
+ * for running migrations within an app
+ */
+class CakeManager extends Manager
+{
+
+    /**
+     * Reset the migrations stored in the object
+     *
+     * @return void
+     */
+    public function resetMigrations()
+    {
+        $this->migrations = null;
+    }
+
+    /**
+     * Prints the specified environment's migration status.
+     *
+     * @param string $environment
+     * @param null $format
+     * @return array Array of migrations
+     */
+    public function printStatus($environment, $format = null)
+    {
+        $migrations = [];
+        if (count($this->getMigrations())) {
+            $env = $this->getEnvironment($environment);
+            $versions = $env->getVersions();
+
+            foreach ($this->getMigrations() as $migration) {
+                if (in_array($migration->getVersion(), $versions)) {
+                    $status = 'up';
+                    unset($versions[array_search($migration->getVersion(), $versions)]);
+                } else {
+                    $status = 'down';
+                }
+
+                $migrations[] = [
+                    'status' => $status,
+                    'id' => $migration->getVersion(),
+                    'name' => $migration->getName()
+                ];
+            }
+
+            foreach ($versions as $missing) {
+                $migrations[] = ['status' => 'up', 'id' => $missing, 'name' => false];
+            }
+        }
+
+        if ($format === 'json') {
+            $migrations = json_encode($migrations);
+        }
+
+        return $migrations;
+    }
+
+    /**
+     * Checks if the migration with version number $version as already been mark migrated
+     *
+     * @param int|string $version Version number of the migration to check
+     * @return bool
+     */
+    public function isMigrated($version)
+    {
+        $adapter = $this->getEnvironment('default')->getAdapter();
+        $versions = array_flip($adapter->getVersions());
+
+        return isset($versions[$version]);
+    }
+
+    /**
+     * Marks migration with version number $version migrated
+     *
+     * @param int|string $version Version number of the migration to check
+     * @param string $path Path where the migration file is located
+     * @return bool|string A string with the error message or true if success
```

Comment:
Never, it's a mistake. Probably a copy / paste from another method (from a previous version of code).